narwhals/_spark_like/dataframe.py

```diff
@@ -142,43 +142,53 @@ def _with_native(self, df: SQLFrameDataFrame) -> Self:
             implementation=self._implementation,
         )
 
+    def _to_arrow_schema(self) -> pa.Schema:  # pragma: no cover
```

Comment:
Reason for being a method rather than a util function is that we need to access a bunch of instance states